clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -51,9 +52,12 @@
     private final ProducerRecord<byte[], byte[]> record1 = new ProducerRecord<>(topic, "key1".getBytes(), "value1".getBytes());
     private final ProducerRecord<byte[], byte[]> record2 = new ProducerRecord<>(topic, "key2".getBytes(), "value2".getBytes());
 
+    @After
+    public void close() {
+        producer.close();
```

Comment:
I submitted an update, in which I removed the global variable and used `try-with-resources` blocks when it was possible. The tests now seem to have a consistent pattern.  I hope I did not misunderstand your comment.